From c6d1a707934c2f71454bc616e08d4a7f100e3405 Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Wed, 18 Apr 2018 15:40:25 +0200
Subject: [PATCH 1/1] efi_loader: selftest: Do not build relocation tests for
 risc-v

The relocation selftest doesn't compile for me on RISC-V. Disable for now.

Signed-off-by: Alexander Graf <agraf@suse.de>
---
 lib/efi_selftest/Makefile | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/lib/efi_selftest/Makefile b/lib/efi_selftest/Makefile
index 0e4980c8a0..40c838a287 100644
--- a/lib/efi_selftest/Makefile
+++ b/lib/efi_selftest/Makefile
@@ -36,9 +36,10 @@ ifeq ($(CONFIG_BLK)$(CONFIG_PARTITIONS),yy)
 obj-$(CONFIG_CMD_BOOTEFI_SELFTEST) += efi_selftest_block_device.o
 endif
 
-# TODO: As of v2018.01 the relocation code for the EFI application cannot
-# be built on x86_64.
+# TODO: As of v2018.05 the relocation code for the EFI application cannot
+# be built on x86_64 / RISC-V.
 ifeq ($(CONFIG_X86_64),)
+ifeq ($(CONFIG_RISCV),)
 
 ifneq ($(CONFIG_CMD_BOOTEFI_SELFTEST),)
 
@@ -64,6 +65,8 @@ $(obj)/efi_selftest_startimage_exit.o: $(obj)/efi_miniapp_file_image_exit.h
 
 $(obj)/efi_selftest_startimage_return.o: $(obj)/efi_miniapp_file_image_return.h
 
-endif
+endif # CONFIG_CMD_BOOTEFI_SELFTEST
 
-endif
+endif # !CONFIG_RISCV
+
+endif # !CONFIG_X86_64
-- 
2.17.0

